env_spec.py

```diff
@@ -40,58 +40,66 @@ def check_type(input_type):
     return True
 
 
-def render_env_var_spec(env_var):
+def render_env_var_spec(env_var_field, env_var_type):
     """
-    Creates html sting from env_variable (field, type). Returns the string.
+    Creates html string from env_variables (field, type). Returns the string.
     """
-    try:
-        env_var_field, env_var_type = env_var.split(": ")
-        env_var_field_lower = env_var_field.lower()
+    env_var_field_lower = env_var_field.lower()
 
+    if env_var_type != "":
         ret_str = (
             f'<label for="env_spec_{env_var_field_lower}"> {env_var_field}</label>'
         )
         ret_str += "\n"
         ret_str += f'<input id="env_spec_{env_var_field_lower}" name="{env_var_field_lower}" />'
         ret_str += "\n"
         ret_str += f'<input id="env_spec_{env_var_field_lower}" name="{env_var_field_lower}" type="{env_var_type}" />'
-    except:
-        env_var_field = env_var
-        env_var_field_lower = env_var_field.lower()
-
+    else:
         ret_str = (
             f'<label for="env_spec_{env_var_field_lower}"> {env_var_field}</label>'
         )
 
     return ret_str
 
 
-def render_env_spec_to_html(input_line):
-    try:
-        input_field, input_type = input_line.split(": ")
-
-        if check_input(input_field) and check_type(input_type):
-            html_output = render_env_var_spec(input_line)
-
-            return html_output
-    except:
-        input_field = input_line
-
-        if check_input(input_field):
-            html_output = render_env_var_spec(input_field)
-
+def render_env_spec_to_html(input_str):
+    """
+    Takes the whole string and splits it first by \n and then by :, if possible. Returns html output or "".
+    """
+    html_output = ""
+    last_string_flag = False
+
+    while 1:
```

Comment:
We will have to ditch `while 1` here. I am terrified of them 😱.

Would something like the following work instead?

```python
lines = input_str.split('\n')

for line in lines:
    # do stuff here
```